src/compose/service-manager.coffee

```diff
@@ -0,0 +1,278 @@
+		.filter((currentService) -> currentService.isSameContainer(service))
+		.get(0)
+
+	getStatus: =>
+		@getAll()
+		.then (services) =>
+			status = _.clone(@volatileState)
+			for service in services
+				status[service.containerId] ?= _.pick(service, [ 'appId', 'imageId', 'status', 'releaseId', 'commit' ])
+			return _.values(status)
+
+	getByDockerContainerId: (containerId) =>
+		@docker.getContainer(containerId).inspect()
+		.then (container) ->
+			if !container.Config.Labels['io.resin.supervised']?
+				return null
+			return Service.fromContainer(container)
+
+	waitToKill: (service, timeout) ->
+		pollInterval = 100
+		timeout = checkInt(timeout, positive: true) ? 60000
+		deadline = Date.now() + timeout
+
+		killmePath = service.killmeFullPathOnHost()
+
+		wait = ->
+			fs.statAsync(killmePath)
+			.then ->
+				fs.unlinkAsync(killmePath).catch(_.noop)
+			.catch (err) ->
+				if Date.now() < deadline
+					Promise.delay(pollInterval).then(wait)
+		wait()
+
+	prepareForHandover: (service) =>
+		@get(service)
+		.then (svc) =>
+			container = @docker.getContainer(svc.containerId)
+			container.update(RestartPolicy: {})
+			.then ->
+				container.rename(name: "old_#{service.serviceName}_#{service.imageId}_#{service.releaseId}")
+
+	updateMetadata: (service, { imageId, releaseId }) =>
+		@get(service)
+		.then (svc) =>
+			@docker.getContainer(svc.containerId).rename(name: "#{service.serviceName}_#{imageId}_#{releaseId}")
+
+	handover: (currentService, targetService) =>
+		# We set the running container to not restart so that in case of a poweroff
+		# it doesn't come back after boot.
+		@prepareForHandover(currentService)
+		.then =>
+			@start(targetService)
+		.then =>
+			@waitToKill(currentService, targetService.labels['io.resin.update.handover-timeout'])
+		.then =>
+			@kill(currentService)
+
+	listenToEvents: =>
+		if @listening
+			return
+		@listening = true
+		@docker.getEvents(filters: type: [ 'container' ])
+		.then (stream) =>
+			stream.on 'error', (err) ->
+				console.error('Error on docker events stream:', err, err.stack)
+			parser = JSONStream.parse()
+			parser.on 'data', (data) =>
+				if data?.status in ['die', 'start']
+					@getByDockerContainerId(data.id)
+					.catchReturn(NotFoundError, null)
+					.then (service) =>
+						if service?
+							if data.status == 'die'
+								@emit('change')
+								@logger.logSystemEvent(logTypes.serviceExit, { service })
+								@containerHasDied[data.id] = true
+							else if data.status == 'start' and @containerHasDied[data.id]
+								@emit('change')
+								delete @containerHasDied[data.id]
+								@logger.logSystemEvent(logTypes.serviceRestart, { service })
+								@logger.attach(@docker, data.id, service.serviceId)
+					.catch (err) ->
+						console.error('Error on docker event:', err, err.stack)
+					return
+			new Promise (resolve, reject) ->
+				parser
+				.on 'error', (err) ->
+					console.error('Error on docker events stream', err)
```

Comment:
console.error

Reply:
Yup, it's on purpose, this should be rare and we'd like to see it in supervisor logs if it happens